tests/endpoint.rs

```diff
@@ -0,0 +1,126 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+
+#![cfg(feature = "remote_endpoint")]
+
+mod subscribers;
+
+use std::any::Any;
+use std::result;
+use std::sync::atomic::{AtomicBool, AtomicU64, Ordering};
+use std::sync::Arc;
+use std::thread;
+use std::time::Duration;
+
+use event_manager::{self, EventManager, MutEventSubscriber, SubscriberId};
+use subscribers::CounterSubscriber;
+
+trait SpecialSubscriber: MutEventSubscriber {
+    fn as_mut_any(&mut self) -> &mut dyn Any;
+}
+
+impl SpecialSubscriber for CounterSubscriber {
+    fn as_mut_any(&mut self) -> &mut dyn Any {
+        self
+    }
+}
+
+#[test]
+fn test_endpoint() {
```

Comment:
Can you also add a comment about what the test exemplifies? For example something in the lines of: showcasing how to use the remote endpoint and fetching the actual type of the subscriber once it was added to event manager.